service/refund.go

```diff
@@ -72,3 +74,61 @@ func (service *RefundService) CreateRefund(req *http.Request, id string, createR
 
 	return paymentSession, &refundResource, Success, nil
 }
+
+// UpdateRefund checks refund status in GovPay and if status is successful saves it to payment object in mongo
+func (service *RefundService) UpdateRefund(req *http.Request, paymentId string, refundId string) (*models.RefundResponse, ResponseType, error) {
+	paymentSession, response, err := service.PaymentService.GetPaymentSession(req, paymentId)
+	if err != nil {
+		err = fmt.Errorf("error getting payment resource: [%v]", err)
+		log.ErrorR(req, err)
+		return nil, response, err
+	}
+
+	if response == NotFound {
+		err = fmt.Errorf("error getting payment resource")
+		log.ErrorR(req, err)
+
+		return nil, NotFound, err
+	}
+
+	index, err := getRefundIndex(paymentSession.Refunds, refundId)
+
+	if err != nil {
+		log.ErrorR(req, err)
+		return nil, NotFound, err
+	}
+	// Get RefundStatus from GovPay to check the status of the refund
+	govPayStatusResponse, response, err := service.GovPayService.GetGovPayRefundStatus(paymentSession, refundId)
+	if err != nil {
+		err = fmt.Errorf("error getting refund status from govpay: [%v]", err)
+		log.ErrorR(req, err)
+		return nil, response, err
+	}
+
+	if govPayStatusResponse.Status == RefundsStatusSuccess {
```

Comment:
what about error status? Surely the db should also get updated rather than sitting in submitted?

Reply:
Done